plugin/src/main/java/org/autorefactor/refactoring/rules/SwitchRefactoring.java

```diff
@@ -90,23 +122,33 @@ public String toString() {
         }
     }
 
-    private static final class RewrittenCase {
+    private static final class SwitchCaseSection {
         /** Must resolve to constant values. */
         private List<Expression> expressions;
-        private Statement stmt;
-
-        public RewrittenCase(List<Expression> expressions, Statement stmt) {
-            this.expressions = expressions;
-            this.stmt = stmt;
+        private List<SwitchCase> existingCases;
+        private List<Statement> stmts;
+
+        public SwitchCaseSection(List<Expression> expressionList, List<SwitchCase> caseList,
+                List<Statement> statementList) {
+            this.expressions = expressionList;
+            this.stmts = statementList;
+            this.existingCases = caseList;
         }
 
         @Override
         public String toString() {
             final StringBuilder sb = new StringBuilder();
-            for (Expression expr : expressions) {
-                sb.append("case ").append(expr).append(":\n");
+            if (expressions != null) {
```

Comment:
It is forbidden to do null checks on collections :)
See Effective Java 2nd Edition by Joshua Bloch:
> Item 43: Return empty arrays or collections, not nulls

Same comment 4 lines down below.